### PR TITLE
feat: Add a cloud-node-manager sidecar container called health-probe-proxy to transfer the traff…

### PR DESCRIPTION
**Pull request by nilo19**
…ic from port 10356 to the kube-proxy health check server port 10256. This sidecar will parse the proxy protocol packet data unit when the user uses private link service integrated with the service annotation and set the proxy protocol annotation. In this case the kube-proxy health probe server will not read the health probe request from the SLB and fails the health check.

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

feat: Add a daemonset called health-probe-proxy to transfer the traffic from port 10356 to the kube-proxy health check server port 10256. This daemonset will parse the proxy protocol packet data unit when the user uses private link service integrated with the service annotation and set the proxy protocol annotation. In this case the kube-proxy health probe server will not read the health probe request from the SLB and fails the health check.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
feat: Add a cloud-node-manager sidecar container called health-probe-proxy to transfer the traffic from port 10356 to the kube-proxy health check server port 10256. This sidecar will parse the proxy protocol packet data unit when the user uses private link service integrated with the service annotation and set the proxy protocol annotation. In this case the kube-proxy health probe server will not read the health probe request from the SLB and fails the health check.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: health-probe-proxy/example.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
```
Comment (feiskyer) [inline health-probe-proxy/example.yaml:1]: let's add the deployment charts into helm charts as part of cnm deployment (a new sidecar for it). And please add a new helm config to enable this sidecar, which is disable by default during preview.

**Comment by nilo19** [comment]
testing windows

File: helm/cloud-provider-azure/values.yaml [hunk 2]
```diff
@@ -77,3 +77,8 @@ cloudNodeManager:
     requestsMemWin: "50Mi"
     limitsCPUWin: "2"
     limitsMemWin: "512Mi"
+  enableHealthProbeProxy: false
+  healthCheckPort: 10356
+  targetPort: 10256
+  healthProbeProxyImage: "ss104301/health-probe-proxy:latest"
```
Comment (nilo19) [inline helm/cloud-provider-azure/values.yaml:83]: Use my personal image as a placeholder and will replace to the mcr image when they are ready.

File: health-probe-proxy/example.yaml (hunk 1 above)
Comment (nilo19) [reply health-probe-proxy/example.yaml:1]: done

File: health-probe-proxy/example.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+kind: DaemonSet
+metadata:
+  name: health-probe-proxy
+  namespace: kube-system
+  labels:
+    component: health-probe-proxy
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (jwtty) [inline health-probe-proxy/example.yaml:9]: remove this line?

Comment (nilo19) [reply health-probe-proxy/example.yaml:9]: This is useful when using addon manager to deploy health probe proxy.

**Comment by nilo19** [comment]
@jwtty can you help approve?

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**